src/nat/profiler/parameter_optimization/prompt_optimizer.py

```
@@ -0,0 +1,419 @@
+                return curr_pop[-1]
+
+            # Produce offspring
+            needed = pop_size - len(next_population)
+            offspring: list[Individual] = []
+            for _ in range(max(0, offspring_size), needed):
+                pass  # ensure bound correctness
+            while len(offspring) < needed:
+                p1 = _select_parent(population)
+                p2 = _select_parent(population)
+                if p2 is p1 and len(population) > 1:
+                    p2 = random.choice([ind for ind in population if ind is not p1])
```

Ruff (0.12.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)